### DISCUSSION TITLE
6.2a-pre won't configure for me

### DISCUSSION THREAD

**Discussion by brianharvey** [jrincayc/ucblogo-code#80] [Q&A]
Here's what happens:
```
...
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking enable_objects... no
checking enable_gitid... no
./configure: line 4362: WX_CONFIG_OPTIONS: command not found
./configure: line 4364: syntax error near unexpected token `3.0.0,'
./configure: line 4364: `WX_CONFIG_CHECK(3.0.0, wxWin=1)'
```

I'm on MacOS 10.13.4.

**Comment by jrincayc** [comment]
Thanks, I'll check mac and windows builds before I make the final release.  I'll also make sure that the error message when wx-config is missing is better.

**Comment by dmalec** [comment]
From a purely mechanical perspective - I've found I need to do a bootstrap step which includes installing wxWidgets system-wide so that autotools has access to the `WX_CONFIG_OPTIONS` macro. My approach on my mac (10.15.7) has been:
```bash
brew install autoconf-archive automake wxwidgets
```

The build itself does then use whichever version of wxWidgets I'm pointing to with the `--with-wx-config=` option.

Less mechanically, I wonder if it's fair game to add that particular macro to a third party macros folder when producing the ucblogo source distribution?

**Reply by jrincayc** [reply]
Hm, I wonder if I can get autotools to do something intelligent when WX_CONFIG_OPTIONS is not available.

**Comment by jrincayc** [comment]
@brianharvey Hm, which commands did you run before running configure and what configure setting did you do?
For me, on the macos 10.11 this worked:
```
tar -xvzf Downloads/ucblogo-6.2pre.tar.gz
cd ucblogo-6.2pre/
./configure --prefix=$HOME/ucblogo_install --with-wx-config=$HOME/wxWidgets-3.0.4/build-cocoa-static/wx-config
make ucblogo.dmg
```
I need to try with the newer wxWidgets.  (Also, getting version 6.1 or 6.2 when it is done in homebrew would be useful.)

**Reply by brianharvey** [reply]
In README.md it says this:

To build Logo under *nix, do this:
```
	autoreconf --install
	./configure
	make
```

so that's what I did!  I just tried your way and still get an error:

```
...
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking enable_objects... no
checking enable_gitid... no
./configure: line 4362: WX_CONFIG_OPTIONS: command not found
./configure: line 4364: syntax error near unexpected token `3.0.0,'
./configure: line 4364: `WX_CONFIG_CHECK(3.0.0, wxWin=1)'
```

I suggest that either the README.md include information on downloading and installing wxWidgets or that wxWidgets be included in the distribution.

**Reply by jrincayc** [reply]
Okay, thanks for the answer.  I agree, the README needs to mention wxWidgets better.

**Comment by jrincayc** [comment]
I changed the configure.ac to handle lack of WX_CONFIG_CHECK better. Pull request #84

I will also change the README to note that installing wxwidgets should be done before autoconf.

**Reply by dmalec** [reply]
Ah, nice on `m4_ifdef`, hadn't known about that previously
